### DISCUSSION TITLE
Add support for AGV in the scheduling environment.

### DISCUSSION THREAD

**Discussion by lshengjian** [ai4co/rl4co#207] [Ideas]
Especially in the FJSP environment with AGVs.

**Comment by fedebotu** [comment]
Do you have any literature/code references for such an environment?

**Comment by lshengjian** [comment]
> Do you have any literature/code references for such an environment?

https://github.com/lshengjian/jsplab/

**Comment by fedebotu** [comment]
Thanks for sharing. I don't have much bandwidth at the moment, so unfortunately, I cannot implement such an environment from scratch at the moment :( Maybe @LTluttmann since he knows much more than me in scheduling, although also pretty busy doing PhD ...

This said, your code seems pretty clean! It should be more than possible to create an environment [similarly to our FJSP](https://github.com/ai4co/rl4co/blob/main/rl4co/envs/scheduling/fjsp/env.py), which is much faster than the `gym` implementations we checked so far. I invite you to contribute to your environment, we will be more than happy to help! :)

**Comment by lshengjian** [comment]
> Thanks for sharing. I don't have much bandwidth at the moment, so unfortunately, I cannot implement such an environment from scratch at the moment :( Maybe @LTluttmann since he knows much more than me in scheduling, although also pretty busy doing PhD ...
> 
> This said, your code seems pretty clean! It should be more than possible to create an environment [similarly to our FJSP](https://github.com/ai4co/rl4co/blob/main/rl4co/envs/scheduling/fjsp/env.py), which is much faster than the `gym` implementations we checked so far. I invite you to contribute to your environment, we will be more than happy to help! :)

Thank you for your invitation. but I'm sorry I can't fully understand your FJSP  environment.

**Comment by fedebotu** [comment]
I see, please let us know any way we could help you :)
